pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -78,11 +90,106 @@ func (r *ReconcileStorageCluster) newOBCStorageClass(initData *ocsv1.StorageClus
 	return retSC
 }
 
+// ensureExternalClusterConfig ensures that Storagclass for the external cluster exist
+func (r *ReconcileStorageCluster) ensureExternalClusterConfig(instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	// check for the status boolean value accepted or not
+	if instance.Status.ExternalSecretFound {
+		return nil
+	}
+	found := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: instance.Namespace,
+		},
+	}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: found.Name, Namespace: found.Namespace}, found)
+	if err != nil {
+		return err
+	}
+	ownerRef := metav1.OwnerReference{
+		UID:        instance.UID,
+		APIVersion: instance.APIVersion,
+		Kind:       instance.Kind,
+		Name:       instance.Name,
+	}
+	var data []ocsv1.ExternalResources
+	err = json.Unmarshal(found.Data[externalClusterSecretDataKeyName], &data)
+	if err != nil {
+		return fmt.Errorf("could not parse json blob, got %+v", err)
+	}
+	for _, d := range data {
+		switch d.Kind {
+		case "ConfigMap":
+			cm := &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            d.Name,
+					Namespace:       d.Name, // TODO: What will be namespace given for ConfigMap?. d.Name or instance.Namespace?.
+					OwnerReferences: []metav1.OwnerReference{ownerRef},
+				},
+			}
+			cm.Data = d.Data
+			found := &corev1.ConfigMap{}
+			err := r.client.Get(context.TODO(), types.NamespacedName{Name: d.Name, Namespace: instance.Namespace}, found)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					reqLogger.Info("Creating Ceph ConfigMap:	" + cm.Name)
+					err = r.client.Create(context.TODO(), cm)
+					if err != nil {
+						return err
+					}
+				}
+				return err
+			}
+		case "Secret":
+			s := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            d.Name,
+					Namespace:       d.Name, // TODO: What will be namespace for Secret?. d.Name or instance.Namespace?.
+					OwnerReferences: []metav1.OwnerReference{ownerRef},
+				},
+			}
+			for k, v := range d.Data {
+				s.Data[k] = []byte(v)
+			}
+			found := &corev1.Secret{}
+			err := r.client.Get(context.TODO(), types.NamespacedName{Name: d.Name, Namespace: instance.Namespace}, found)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					reqLogger.Info("Creating Ceph Secret: 	" + s.Name)
+					err = r.client.Create(context.TODO(), s)
+					if err != nil {
+						return err
+					}
+				}
+				return err
+			}
+		case "StorageClass":
```

Comment:
My understanding is that there is not *one* resource of kind `StorageClass`, but three, one for each actual StorageClass that will be created. Also, you do not want to do this here in this function. In this switch case, you only want to add the resource `d` to a new `[]ExternalResource` list defined outside the `for _, d := range data` loop. The rest you'll want to do in `newStorageClasses()`.

Reply:
> My understanding is that there is not _one_ resource of kind `StorageClass`, but three, one for each actual StorageClass that will be created. Also, you do not want to do this here in this function. In this switch case, you only want to add the resource `d` to a new `[]ExternalResource` list defined outside the `for _, d := range data` loop. The rest you'll want to do in `newStorageClasses()`.

When you say "The rest", you meant the `case` just for the StorageClass right?. Meaning that we should remove that and create as you said?. IIRC, we need the switch Case for creation of the  `Secrets` and `ConfigMaps`.

Reply:
See below.

Reply:
We're still following the switch cases approach as modifying the `newStorageClasses ` will I believe does not make much sense as that function job is to just create StorageClasses and nothing else.